### ISSUE TITLE
SD card leads to assert error for 101st log file

### ISSUE THREAD

**Issue by dominiknatter** [bitcraze/crazyflie-firmware#988]
Hi,

I have observed a high accumulation of these errors today:
SYS: Assert failed at crazyflie-firmware/vendor/FreeRTOS/tasks.c:1856

More precisely, they occurred every time I tried to switch the usd.logging param to 0 (I'm using the "enable on startup" flag). 
At first I thought it's just another assert error with SD cards (see our discussions [here](https://forum.bitcraze.io/viewtopic.php?t=5221)). However, I have then realized that my SD card contains logs from 0 to 99 now. When deleting the file with suffix 99, this error vanishes for one logging procedure, and then arises again for the next.

So in conclusion it seems this error occurs when the system cannot write any 2-digit suffix to the log filename anymore.

Is allowing for only 100 log files (aka using only 2-digit suffixes) a dedicated, desired choice from your side or could this be changed? Either way, I thought having this error documented could be useful, given that the error message is not very straightforwardly pointing to this cause.

Best,
Dominik

**Comment by krichardsson** [comment]
Yes, you are correct. As we don't have proper string handling functions the file name only supports 2 chars for number, that is a limit of 100 files. If this limitation is not acceptable, the easiest solution is probably to add one more char to get a maximum of 1000 files.

You can find (some of) the related code here https://github.com/bitcraze/crazyflie-firmware/blob/2022.01/src/deck/drivers/src/usddeck.c#L579-L581 and here https://github.com/bitcraze/crazyflie-firmware/blob/2022.01/src/deck/drivers/src/usddeck.c#L827-L840
